### PR TITLE
Cleanup ObjectManager usage - Magento_Catalog ViewModel,Plugin

### PR DESCRIPTION
**Pull request by Bartlomiejsz**
### Description (*)

This PR cleanups ObjectManager usage in Magento_Catalog ViewModel and Plugin, by removing object manager instantiation from non-api classes. This is first part of https://github.com/magento/magento2/pull/27135 split into smaller chunks to proceed it easily.

### Fixed Issues (if relevant)

N/A

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Catalog/Test/Unit/Plugin/Model/ResourceModel/ConfigTest.php [hunk 1]
```diff
@@ -3,33 +3,49 @@
  * Copyright © Magento, Inc. All rights reserved.
  * See COPYING.txt for license details.
  */
+declare(strict_types=1);
 
 namespace Magento\Catalog\Test\Unit\Plugin\Model\ResourceModel;
 
+use Magento\Catalog\Model\ResourceModel\Config as Subject;
```
Comment (lbajsarowicz) [inline app/code/Magento/Catalog/Test/Unit/Plugin/Model/ResourceModel/ConfigTest.php]: Avoid using subject as it's misleading.
Use ConfigResourceModel instead

File: app/code/Magento/Catalog/Test/Unit/Plugin/Model/ResourceModel/ConfigTest.php [hunk 2]
```diff
@@ -3,33 +3,49 @@
...
+    /**
+     * @var StateInterface|MockObject
+     */
     private $cacheState;
 
-    /** @var SerializerInterface|\PHPUnit_Framework_MockObject_MockObject */
+    /**
+     * @var SerializerInterface|MockObject
+     */
     private $serializer;
```
Comment (lbajsarowicz) [inline app/code/Magento/Catalog/Test/Unit/Plugin/Model/ResourceModel/ConfigTest.php]: Suffix with 'mock'

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Correct the tested class names (avoid generic aliases). 
Add 'mock' suffix to mocked properties in unit tests.

**Comment by Bartlomiejsz** [comment]
@lbajsarowicz applied fixes for review comments, please review

**Review by lbajsarowicz — APPROVED** [review]
:heavy_check_mark: Thank you for your contribution

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-7139](https://jira.corp.magento.com/browse/ENGCOM-7139) has been created to process this Pull Request
:eight_spoked_asterisk: @lbajsarowicz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Comment by engcom-Echo** [comment]
@magento run Functional Tests B2B, @magento run Functional Tests CE, @magento run Functional Tests EE

**Comment by engcom-Echo** [comment]
@magento run Functional Tests CE, @magento run Functional Tests EE

**Comment by engcom-Echo** [comment]
@magento run Functional Tests EE

**Pull request merged**